runtime/gc_vlhgc/VLHGCAccessBarrier.cpp

```diff
@@ -271,6 +271,20 @@ MM_VLHGCAccessBarrier::copyArrayCritical(J9VMThread *vmThread, GC_ArrayObjectMod
 	}
 }
 
+IDATA
+MM_VLHGCAccessBarrier::indexableDataDisplacement(J9VMThread *vmThread,  J9IndexableObject *src,  J9IndexableObject *dst)
+{
+	IDATA displacement = 0;
+	GC_ArrayObjectModel *indexableObjectModel = &_extensions->indexableObjectModel;
+
+	Assert_MM_true((0 == indexableObjectModel->getSizeInElements(dst)) || indexableObjectModel->isInlineContiguousArraylet(dst));
```

Comment:
I believe we can put a stronger assertion, that this API will not be used if real arraylets are enabled, basically not be used:
- running Balanced, with offheap disabled
- running Metronome

so to make this assert cheaper we can combine it with the calculation below:
```
If offheap enabled
   If adjacent 
	displacement = diff
Else
	unreachable 
```

We can also indexableDataDisplacement to RealtimeAccessBarrier which would be simply always unreachable

Reply:
even simpler:

```
assert(offheap enabled)

If adjacent
   displacement = diff
```

also `offheap enabled` can be checked against _extensions->isVirtualLargeObjectHeapEnabled, so no need to even cache indexableObjectModel into a local since it will be used only once